File: Here/app/src/main/java/com/eason/here/model/LoginStatus.java
package com.eason.here.model;

import android.content.Context;

/**
 * 用户的登录状态
 *
 * Created by Eason on 8/28/15.
 */
public class LoginStatus {

    private static User user;

    private static String pushKey = "";

    private static Context context;

    public static User getUser() {
        return user;
    }

    public static void setUser(User user) {
        LoginStatus.user = user;
    }

    public static String getPushKey() {
        return pushKey;
    }

    public static void setPushKey(String pushKey) {
        LoginStatus.pushKey = pushKey;
    }

    public static void logout(){
        user = null;
    }

    public static void init(Context context){
        LoginStatus.context = context;
    }

    /**
     * 检测当前是否是用户登录模式,如果是登录模式，返回true
     *
     * @return
     */
    public static boolean getIsUserMode(){
        if (user!=null){
            return true;
        }else{
            return false;
        }
    }

}

File: Marker-iOS/LoveMarker/LoveMarker/Model/LoginStatus.m
//
//  LoginStatus.m
//  LoveMarker
//
//  Created by BigHead_Chen on 5/23/16.
//  Copyright © 2016 Eason. All rights reserved.
//

#import "LoginStatus.h"
#import "CoreDataUser.h"
#import "MyDataController.h"
#import "HttpRequest.h"
#import "ErrorState.h"
#import "NSObject+ObjectMap.h"
#import "ResponseResult.h"

@implementation LoginStatus

__strong static id instance = nil;

+(void)initInstanc{
    if (!instance) {
        instance = [[self alloc] init];
    }
}

+(instancetype)getInstance{
    if (!instance) {
        instance = [[self alloc] init];
    }
    
    return instance;
}

-(Boolean)getIsUserModel{
    if (self.user) {
        return YES;
    }else{
        return NO;
    }
}

-(void)logout{
    [[[MyDataController alloc]init] saveOrUpdataUserCoreDataWithUsername:self.user.username password:self.user.password key:KEY_VALUE isLogout:YES];
    self.user = nil;
}

-(void)loginWithUsername:(NSString*)username password:(NSString*)password pushKey:(NSString*)pushKey successHandler:(LoginSuccessHandler)successHandler
           failedHandler:(LoginFailedHandler)failedHandler{
    
    [HttpRequest loginWithUsername:username password:password pushKey:pushKey responseData:^(ResponseResult *response, NSObject *resultObject) {
        ResponseResult* result = (ResponseResult*)response;
        if (resultObject!=nil&&(result.status) == Error_Code_Correct) {
            User* user = [NSObject objectOfClass:@"User" fromJSON:(NSDictionary*)resultObject];
            [[LoginStatus getInstance] setUser:user];
            [[[MyDataController alloc]init] saveOrUpdataUserCoreDataWithUsername:user.username password:user.password key:KEY_VALUE isLogout:NO];
            NSArray *result = [[[MyDataController alloc]init] getUserCoreDataWithUsername:user.username];
            successHandler();
            NSLog(@"fetch result :%@",result);
            
        }else if((result.status) == Error_Code_User_Not_Found){
            failedHandler(result.status);
        }else if ((result.status) == ERROR_CODE_USER_OR_PASSWORD_INVALID){
            failedHandler(result.status);
        }
        
    }];
}

-(void)autoLogin{
    CoreDataUser* user = [[[[MyDataController alloc] init] getUserCoreDataWithDefualKey] firstObject];
    if (user && !user.isLogout) {
        [HttpRequest loginWithUsername:user.username password:user.password pushKey:@"" responseData:^(NSObject *response, NSObject *resultObject) {
            ResponseResult* result = (ResponseResult*)response;
            if (resultObject!=nil&&(result.status) == Error_Code_Correct) {
                User* user = [NSObject objectOfClass:@"User" fromJSON:(NSDictionary*)resultObject];
                [[LoginStatus getInstance] setUser:user];
                [[[MyDataController alloc]init] saveOrUpdataUserCoreDataWithUsername:user.username password:user.password key:KEY_VALUE isLogout:NO];
                NSLog(@"aotu login success");
                
            }else{
                
            }
        }];
    }
}

@end
